core/DbHelper.php

```diff
@@ -37,7 +37,7 @@ public static function getTablesInstalled($forceReload = true)
      */
     public static function tableExists($tableName)
     {
-        return Db::get()->query("SHOW TABLES LIKE ?", $tableName)->rowCount() > 0;
+        return Db::get()->query(sprintf("SHOW TABLES LIKE '%s'", $tableName))->rowCount() > 0;
```

Comment:
could maybe add a comment to the function to not use user input as table name to prevent SQL injections but that kind of applies everywhere for tables in Matomo anyway. At the same time might not hurt to mention it

Reply:
Could also just remove `'` characters from $tableName?

Reply:
this alone would probably not prevent injections. wouldn't hurt though to do it additionally.

Reply:
out of curiosity, what kind of value would get past that? I guess we could also remove `'";\`` characters?

Reply:
@diosmosis it depends on the use cases and how it's used. Eg you could trick Matomo into thinking the table is there or not there, or find out what tables exist by changing the input, etc. Make use of `_` and `%`...

Reply:
👍 that makes sense.

Reply:
updated.